terra_notebook_utils/drs.py

```diff
@@ -63,51 +66,121 @@ def fetch_drs_info(drs_url: str) -> dict:
+        response_json = resp.json()
+
+        if 'response' in response_json:
+            if 'text' in response_json['response']:
+                error_details = f"Error: {response_json['response']['text']}"
+            else:
+                error_details = ""
+        else:
+            error_details = ""
+
+        raise DRSResolutionError(f"Unexpected response while resolving DRS path. Expected status 200, got "
+                                 f"{resp.status_code}. {error_details}")
+
     return resp_data
 
+def extract_credentials_from_drs_response(response: dict) -> Optional[dict]:
+    if 'googleServiceAccount' not in response or response['googleServiceAccount'] is None:
+        credentials_data = None
+    else:
+        credentials_data = response['googleServiceAccount']['data']
+
+    return credentials_data
+
+def convert_martha_v2_response_to_DRSInfo(drs_url: str, drs_response: dict) -> DRSInfo:
+    """
+    Convert response from martha_v2 to DRSInfo
+    """
+    if 'data_object' in drs_response['dos']:
+        credentials_data = extract_credentials_from_drs_response(drs_response)
+        data_object = drs_response['dos']['data_object']
+
+        if 'urls' not in data_object:
+            raise DRSResolutionError(f"No GS url found for DRS uri '{drs_url}'")
+        else:
+            data_url = None
+            for url_info in data_object['urls']:
+                if 'url' in url_info and url_info['url'].startswith(_GS_SCHEMA):
+                    data_url = url_info['url']
+                    break
+            if data_url is None:
+                raise DRSResolutionError(f"No GS url found for DRS uri '{drs_url}'")
+
+        bucket_name, key = _parse_gs_url(data_url)
+        return DRSInfo(credentials=credentials_data,
+                       bucket_name=bucket_name,
+                       key=key,
+                       name=data_object.get('name'),
+                       size=data_object.get('size'),
+                       updated=data_object.get('updated'))
+    else:
+        raise DRSResolutionError(f"No metadata was returned for DRS uri '{drs_url}'")
+
+def convert_martha_v3_response_to_DRSInfo(drs_url: str, drs_response: dict) -> DRSInfo:
+    """
+    Convert response from martha_v3 to DRSInfo
+    """
+    if 'gsUri' not in drs_response:
+        raise DRSResolutionError(f"No GS url found for DRS uri '{drs_url}'")
+
+    credentials_data = extract_credentials_from_drs_response(drs_response)
+
+    # Related ticket links:
+    # WA-344: Return file name in martha_response (https://broadworkbench.atlassian.net/browse/WA-344)
+    # WA-348: Add this new file name in TNU (https://broadworkbench.atlassian.net/browse/WA-348)
+    return DRSInfo(credentials=credentials_data,
+                   bucket_name=drs_response.get('bucket'),
+                   key=drs_response.get('name'),
+                   name=None,  # currently martha_v3 doesn't return the file name. This should be changed in WA-348.
```

Comment:
FYI- WA-344 is merged, so WA-348 is unblocked if you want to put it in progress and incorporate it into this PR. Or a separate PR works too if this one is close to merging.